### ISSUE TITLE
unreadable filed _filesystem ？ win10 20h2, NodeJS v12.19.0

### ISSUE THREAD

**Issue by SunSeekerX** [cristiammercado/node-disk-info#10]
`const disks = await nodeDiskInfo.getDiskInfo()`

### when i trying to print the disks, i got an unreadable message with filed _filesystem， here is the screen capture.

![image](https://user-images.githubusercontent.com/48562733/97284247-c6c86d00-187b-11eb-8864-85c97c66000e.png)

**Comment by Ekristoffe** [comment]
@SunSeekerX  hello is your OS in Chinese ?

**Comment by SunSeekerX** [comment]
> @SunSeekerX hello is your OS in Chinese ?

Yes.

**Comment by Ekristoffe** [comment]
could you please use this command in a console (CMD in windows):
wmic logicaldisk get Caption,FreeSpace,Size,VolumeSerialNumber,Description  /format:list
and paste the result here (if it is readable)

**Comment by SunSeekerX** [comment]
> could you please use this command in a console (CMD in windows):
> wmic logicaldisk get Caption,FreeSpace,Size,VolumeSerialNumber,Description /format:list
> and paste the result here (if it is readable)

Here is the result in powershell
```❯ wmic logicaldisk get Caption,FreeSpace,Size,VolumeSerialNumber,Description /format:list


Caption=C:
Description=本地固定磁盘
FreeSpace=44054093824
Size=127585816576
VolumeSerialNumber=D41CE609


Caption=D:
Description=本地固定磁盘
FreeSpace=496723611648
Size=500104691712
VolumeSerialNumber=98C83D91


Caption=W:
Description=本地固定磁盘
FreeSpace=164640075776
Size=250058280960
VolumeSerialNumber=1C2D3D72

```

**Comment by Ekristoffe** [comment]
Thanks I will check why it fail

**Comment by Ekristoffe** [comment]
I think I have found the reason:
By default Linux and other Unix based OS have their console output encoded in UTF8.
Windows on the other hand doesn't depending on the OS language.
In order to test a solution may I ask you to modify one file ?
In the file /utils/constants.js there is this line:
`Constants.WINDOWS_COMMAND = 'wmic logicaldisk get Caption,FreeSpace,Size,VolumeSerialNumber,Description  /format:list';`
I have 3 solution (just a small change in the wording:
Solution 1:
`Constants.WINDOWS_COMMAND = 'cmd /u /c wmic logicaldisk get Caption,FreeSpace,Size,VolumeSerialNumber,Description  /format:list';`
Solution 2:
`Constants.WINDOWS_COMMAND = @chcp 65001 >nul & cmd /d/s/c 'wmic logicaldisk get Caption,FreeSpace,Size,VolumeSerialNumber,Description  /format:list';`
Solution 3:
`Constants.WINDOWS_COMMAND = 'cmd /c chcp 65001>nul && wmic logicaldisk get Caption,FreeSpace,Size,VolumeSerialNumber,Description  /format:list';`

The first solution ask CMD to return the output as unicode.
The other 2 change the codepage of CMD to UTF-8.

Could you please test the change.
Thank you.

**Comment by SunSeekerX** [comment]
Neither seems to work.

**Result**
```
❯ node .\demo.mjs
res>>> {
  drivesInfo: [
    Drive {
      _filesystem: '���ع̶�����',
      _blocks: 127585816576,
      _used: 85165109248,
      _available: 42420707328,
      _capacity: '67%',
      _mounted: 'C:'
    },
    Drive {
      _filesystem: '���ع̶�����',
      _blocks: 500104691712,
      _used: 3524648960,
      _available: 496580042752,
      _capacity: '1%',
      _mounted: 'D:'
    },
    Drive {
      _filesystem: '���ع̶�����',
      _blocks: 250058280960,
      _used: 85214474240,
      _available: 164843806720,
      _capacity: '34%',
      _mounted: 'W:'
    }
  ],
  drivesInfo1: [
    Drive {
      _filesystem: '���ع̶�����',
      _blocks: 127585816576,
      _used: 85165109248,
      _available: 42420707328,
      _capacity: '67%',
      _mounted: 'C:'
    },
    Drive {
      _filesystem: '���ع̶�����',
      _blocks: 500104691712,
      _used: 3524648960,
      _available: 496580042752,
      _capacity: '1%',
      _mounted: 'D:'
    },
    Drive {
      _filesystem: '���ع̶�����',
      _blocks: 250058280960,
      _used: 85214474240,
      _available: 164843806720,
      _capacity: '34%',
      _mounted: 'W:'
    }
  ],
  drivesInfo3: [
    Drive {
      _filesystem: '���ع̶�����',
      _blocks: 127585816576,
      _used: 85165109248,
      _available: 42420707328,
      _capacity: '67%',
      _mounted: 'C:'
    },
    Drive {
      _filesystem: '���ع̶�����',
      _blocks: 500104691712,
      _used: 3524648960,
      _available: 496580042752,
      _capacity: '1%',
      _mounted: 'D:'
    },
    Drive {
      _filesystem: '���ع̶�����',
      _blocks: 250058280960,
      _used: 85214474240,
      _available: 164843806720,
      _capacity: '34%',
      _mounted: 'W:'
    }
  ]
}
```

**demo.js**
`
import * as diskInfo from "./dist/index.js";

(async () => {
  try {
    const res = diskInfo.getDiskInfoSync();
    console.log("res>>>", res);
  } catch (error) {
    console.log("error>>>", error.message);
  }
})();

`

**constants**
`
export class Constants {

    /**
     * Command to execute on Windows.
     */
    public static readonly WINDOWS_COMMAND: string = 'wmic logicaldisk get Caption,FreeSpace,Size,VolumeSerialNumber,Description  /format:list';
    public static readonly WINDOWS_COMMAND1: string = 'cmd /u /c wmic logicaldisk get Caption,FreeSpace,Size,VolumeSerialNumber,Description /format:list';
    public static readonly WINDOWS_COMMAND2: string = "@chcp 65001 >nul & cmd /d/s/c 'wmic logicaldisk get Caption,FreeSpace,Size,VolumeSerialNumber,Description /format:list'";
    public static readonly WINDOWS_COMMAND3: string = 'cmd /c chcp 65001>nul && wmic logicaldisk get Caption,FreeSpace,Size,VolumeSerialNumber,Description /format:list';

    /**
     * Command to execute on Linux.
     */
    public static readonly LINUX_COMMAND: string = 'df -P | awk \'NR > 1\'';

    /**
     * Command to execute on OSX.
     */
    public static readonly DARWIN_COMMAND: string = 'df -P | awk \'NR > 1\'';

}
`

**Comment by SunSeekerX** [comment]
I add three function，the Solution 2 cmd runs with exception.
```
❯ node .\demo.mjs
''wmic' is not recognized as an internal or external command,
operable program or batch file.
error>>> Command failed: @chcp 65001 >nul & cmd /d/s/c 'wmic logicaldisk get Caption,FreeSpace,Size,VolumeSerialNumber,Description /format:list'
''wmic' is not recognized as an internal or external command,
operable program or batch file.
```

**Comment by Ekristoffe** [comment]
I think I have found a better one but It may require some testing (sorry for that).
Could you try this version published some minute ago ?
@ekristoffe/node-disk-info 
version 1.2.2-beta.0

Ps there will be a lot happening in the console, this is normal (still in debugging).
Unfortunately I don't have a system to test it ...
If you can sanitize it (remove information you find not appropriate) and paste it here ?
Thanks.

**Comment by SunSeekerX** [comment]
> I think I have found a better one but It may require some testing (sorry for that).
> Could you try this version published some minute ago ?
> @ekristoffe/node-disk-info
> version 1.2.2-beta.0
> 
> Ps there will be a lot happening in the console, this is normal (still in debugging).
> Unfortunately I don't have a system to test it ...
> If you can sanitize it (remove information you find not appropriate) and paste it here ?
> Thanks.

Here is the output
```
❯ node .\demo.mjs
osInfo

CodeSet=936
CountryCode=86




tokens [ 'CodeSet', '936' ]
section CodeSet
data 936
tokens [ 'CountryCode', '86' ]
section CountryCode
data 86
codeset 936
buffer0

Caption=C:
Description=���ع̶�����
FreeSpace=41054773248
Size=127585816576
VolumeSerialNumber=D41CE609


Caption=D:
Description=���ع̶�����
FreeSpace=496599576576
Size=500104691712
VolumeSerialNumber=98C83D91


Caption=W:
Description=���ع̶�����
FreeSpace=165167165440
Size=250058280960
VolumeSerialNumber=1C2D3D72




buffer1 <Buffer 0d 0d 0a 0d 0d 0a 43 61 70 74 69 6f 6e 3d 43 3a 0d 0d 0a 44 65 73 63 72 69 70 74 69 6f 6e 3d fd fd fd 39 36 fd fd fd fd fd 0d 0d 0a 46 72 65 65 53 70 ... 315 more bytes>
buffer2

Caption=C:
Description=���96�����
FreeSpace=41054773248
Size=127585816576
VolumeSerialNumber=D41CE609


Caption=D:
Description=���96�����
FreeSpace=496599576576
Size=500104691712
VolumeSerialNumber=98C83D91


Caption=W:
Description=���96�����
FreeSpace=165167165440
Size=250058280960
VolumeSerialNumber=1C2D3D72




buffer3 <Buffer 0d 0d 0a 0d 0d 0a 43 61 70 74 69 6f 6e 3d 43 3a 0d 0d 0a 44 65 73 63 72 69 70 74 69 6f 6e 3d ef bf bd ef bf bd ef bf bd 39 36 ef bf bd ef bf bd ef bf ... 363 more bytes>
buffer4

Caption=C:
Description=���96�����
FreeSpace=41054773248
Size=127585816576
VolumeSerialNumber=D41CE609


Caption=D:
Description=���96�����
FreeSpace=496599576576
Size=500104691712
VolumeSerialNumber=98C83D91


Caption=W:
Description=���96�����
FreeSpace=165167165440
Size=250058280960
VolumeSerialNumber=1C2D3D72
```

**Comment by Ekristoffe** [comment]
Strange it seems the output of wmic is not using the os set codepage.
What is the output of this command:
`chcp`
in cmd ?

**Comment by Ekristoffe** [comment]
i made a second beta with 2 different log inside:
one log will start with 
`chcp: `
the other one with
`wmic: `
Does one of those 2 display correctly the data ?

ex:
```
chcp: UTF-8, 
Caption=E:
Description=Removable Disk
FreeSpace=9533808640
Size=15481176064
VolumeName=フリー
VolumeSerialNumber=6453467B
```
vs
```
wmic: cp932, 
Caption=E:
Description=Removable Disk
FreeSpace=9533808640
Size=15481176064
VolumeName=繝輔Μ繝ｼ
VolumeSerialNumber=6453467B
```

**Comment by SunSeekerX** [comment]
> Strange it seems the output of wmic is not using the os set codepage.
> What is the output of this command:
> `chcp`
> in cmd ?

```
C:\Users\ssx>chcp
活动代码页: 936
```

**Comment by Ekristoffe** [comment]
I just published the version 1.2.2-beta.2
https://www.npmjs.com/package/@ekristoffe/node-disk-info/v/1.2.2-beta.2
I really hope this will work on your system.

**Comment by SunSeekerX** [comment]
> I just published the version 1.2.2-beta.2
> https://www.npmjs.com/package/@ekristoffe/node-disk-info/v/1.2.2-beta.2
> I really hope this will work on your system.

It worked!
```
❯ node .\demo.mjs
res>>> [
  Drive {
    _filesystem: '本地固定磁盘',
    _blocks: 127585816576,
    _used: 86505521152,
    _available: 41080295424,
    _capacity: '68%',
    _mounted: 'C:'
  },
  Drive {
    _filesystem: '本地固定磁盘',
    _blocks: 500104691712,
    _used: 3460202496,
    _available: 496644489216,
    _capacity: '1%',
    _mounted: 'D:'
  },
  Drive {
    _filesystem: '本地固定磁盘',
    _blocks: 250058280960,
    _used: 85365428224,
    _available: 164692852736,
    _capacity: '34%',
    _mounted: 'W:'
  }
]
```

**Comment by Ekristoffe** [comment]
Great !!!
I will do a proper PR tomorrow.

**Comment by Ekristoffe** [comment]
#15 done.

**Comment by cristiammercado** [comment]
Hi,

This issue has been fixed in [v1.3.0](https://github.com/cristiammercado/node-disk-info/releases/tag/v1.3.0) already released in [npm](https://www.npmjs.com/package/node-disk-info). I proceed to close it.
